visualizer/layout/collapsed-layout.js

```diff
@@ -23,7 +27,7 @@ class CollapsedLayout {
 
     // TODO: stop relying on coincidental Map.keys() order (i.e. stuff would break when child occurs before parent)
     // e.g. we could attach .topNodes or some iterator to Layout instances
-    this.topLayoutNodes = new Set([...this.layoutNodes.values()].filter(layoutNode => !layoutNode.parent))
+    this.topLayoutNodes = new Set(layoutNodesArray.filter(layoutNode => !layoutNode.parent))
```

Comment:
Use a for(;;) loop here might help. I would avoid the in-between array allocation, adding straight to the Set.